Standardize ACME instruction details

ptgott opened the pull request:
Our Getting Started guides often include instructions for
configuring Let's Encrypt and ACME before starting
Teleport, but not all of these instructions have the same
level of detail, and some are missing some context around
how Teleport uses ACME and why you need to open port 443 on
your Proxy Service host. This change adds an include that
spells out these instructions and invokes the include in the
appropriate guides.

The intention was to include as much relevant information within
the guides themselves to prevent the reader from having to
navigate to other pages.

Closes #6448

ptgott committed:
Standardize ACME instruction details

Our Getting Started guides often include instructions for
configuring Let's Encrypt and ACME before starting
Teleport, but not all of these instructions have the same
level of detail, and some are missing some context around
how Teleport uses ACME and why you need to open port 443 on
your Proxy Service host. This change adds an include that
spells out these instructions and invokes the include in the
appropriate guides.

The intention was to include as much relevant information within
the guides themselves to prevent the reader from having to
navigate to other pages.

Closes #6448

zmb3 commented on docs/pages/application-access/getting-started.mdx:
```diff
@@ -31,31 +31,17 @@ $ docker run -d -p 3000:3000 grafana/grafana
-using Let's Encrypt [ACME](https://letsencrypt.org/how-it-works/) protocol.
-
 We will assume that you have configured DNS records for `teleport.example.com`
 and `*.teleport.example.com` to point to the Teleport node.
```
```suggestion
and `*.teleport.example.com` to point to the Teleport proxy.
```

Right?

zmb3 commented on docs/pages/includes/acme.mdx:
```diff
@@ -0,0 +1,21 @@
+Let's Encrypt verifies that you control the domain name of your Teleport deployment by communicating with the HTTPS server listening on port 443 of your Teleport Proxy Service.
+
+You can configure Teleport to complete the Let's Encrypt verification process—called the ACME protocol—by running the following `teleport configure` command, where `tele.example.com` is the domain name of your Teleport cluster and `user@example.com` is an email address used for notifications (you can use any domain):
```
This reads a little bit awkward to me. My understanding is that `teleport configure` doesn't actually configure anything, but rather spits out some configuration that you can then paste into your teleport.yaml.

The way this reads, it sounds like `teleport configure` is actually modifying the file for you.

zmb3 commented on docs/pages/includes/database-access/start-auth-proxy.mdx:
```diff
@@ -3,24 +3,13 @@ Download the latest version of Teleport for your platform from our
 
 Teleport requires a valid TLS certificate to operate and can fetch one automatically
-using Let's Encrypt [ACME](https://letsencrypt.org/how-it-works/) protocol. We
+using Let's Encrypt. We
```
Strange place for a line break.

ptgott committed:
Respond to PR feedback

- Substitute "proxy" for "node" where it was incorrectly used
- Some small stylistic fixes
- Clarify that "teleport configure" does not write the config
  itself

russjones commented:
@ptgott Looks like this was approved, can we merge?

ptgott committed:
Merge branch 'master' into paul.gottschling/6448-tls-clarity
